lib/exercism/client_version.rb

```diff
@@ -0,0 +1,21 @@
+module ExercismWeb
+  class ClientVersion
+    attr_accessor :user
+
+    def initialize(user:)
+      @user = user
+    end
+
+    def notice_when_client_outdated
+      unless dismissed?
+        "A new version of the exercism client is available (cli-2.4.0). Run <code>exercism upgrade</code> to get the latest hotness!"
```

Comment:
I agree this must stay a short-term thing. I'm just running out of time this week and want to keep it simple, so I can get to the real experiment code. #3416 exists to fix this while the experiment runs. Keep in mind that, in a pinch, we can even just turn the banner off with the feature flag, without having to synchronize a deploy of this repo.

See my edited description for why I guessed at 2.4.0, but I'd be happy to make this 2.3.1 at your discretion.